### PR TITLE
Fixing ip address without mask bug

### PR DESCRIPTION
**Pull request by haddystuff**
##### SUMMARY
This PR was requested in #3776. So this just fork of branch in #3776 but with bugfix(#3768)

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/net_tools/nmcli.py

##### ADDITIONAL INFORMATION
Most of information can be find in #3776. I should mention that i forked this way because this fix depends on adding list support in ipv6 parameter. That's why diff contains diff from #3776.

### CODE REVIEW

File: plugins/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -144,9 +144,10 @@
     ip6:
         description:
             - The IPv6 address to this interface.
-            - Use the format C(abbe::cafe).
+            - Use the format C(abbe::cafe/128) or C(abbe::cafe).
             - If defined and I(method6) is not specified, automatically set C(ipv6.method) to C(manual).
-        type: str
+        type: list
+        elements: str
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:150]: I don't think this part should be in this PR.

File: changelogs/fragments/3768-nmcli_fix_changed_when_no_mask_set.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+bugfixes:
+  - nmcli - fix returning "changed" when no mask set for IPv4 or IPv6 addresses on task rerun
```
Comment (russoz) [inline changelogs/fragments/3768-nmcli_fix_changed_when_no_mask_set.yml:3]:
```suggestion
  - nmcli - fix returning ``changed`` when no mask set for IPv4 or IPv6 addresses on task rerun
```

File: changelogs/fragments/3768-nmcli_fix_changed_when_no_mask_set.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+    (https://github.com/ansible-collections/community.general/issues/3768).
```
Comment (russoz) [inline changelogs/fragments/3768-nmcli_fix_changed_when_no_mask_set.yml:4]: Please end the file with one newline char.

**Review by russoz — COMMENTED** [review]
Hi @haddystuff thanks for the contribution! A couple of adjustments are needed.

**Comment by haddystuff** [comment]
I have rewrited ```enforce_ipv6_cidr_notation``` method so it could work with str. So this PR is now fully independed form #3776.

**Review by felixfontein — APPROVED** [review]
About the changelog fragment: I think all thee versions (no quotes, double quotes, and double backticks) are fine, so I don't care which one you'll take :)

**Review by Ajpantuso — APPROVED** [review]
LGTM as well

**Comment by felixfontein** [comment]
@haddystuff thanks for fixing this!
@russoz @Ajpantuso thanks for reviewing!

**Comment by felixfontein** [comment]
If someone wants this bugfix to appear in the next 3.8.x release, this will need to be backported manually.

**Pull request merged**